src/coreclr/tools/Common/Compiler/DependencyAnalysis/Target_ARM64/ARM64Emitter.cs

```diff
@@ -150,14 +150,22 @@ public void EmitJMP(ISymbolNode symbol)
         {
             if (symbol.RepresentsIndirectionCell)
             {
+                Builder.RequireInitialPointerAlignment();
+
+                if (Builder.CountBytes % Builder.TargetPointerSize == 0)
+                {
```

Comment:
Does this really need to be here? Could we do an adrp/ldr pair for this instead of a [nop] ldr, ldr ....  reloc'd symbol?

Reply:
Opened https://github.com/dotnet/runtime/issues/121129 to track as follow-up work.